core/src/main/java/brooklyn/BrooklynVersion.java

```diff
@@ -19,80 +19,215 @@
+              ManifestHelper mh = Osgis.ManifestHelper.forManifest(u.openStream());
+              if (BROOKLYN_CORE_SYMBOLIC_NAME.equals(mh.getSymbolicName())) {
+                  Attributes attrs = mh.getManifest().getMainAttributes();
+                  for (Object key: attrs.keySet()) {
+                      // key is an Attribute.Name; toString converts to string
+                      versionProperties.put(key.toString(), attrs.getValue(key.toString()));
+                  }
+                  return;
+              }
+          } catch (Exception e) {
+              Exceptions.propagateIfFatal(e);
+              log.warn("Error reading OSGi manifest from "+u+" when determining version properties: "+e, e);
+          }
+      }
+      if (isDevelopmentEnvironment()) {
+          // allowed for dev env
+          log.trace("No OSGi manifest available to determine version properties");
+      } else {
+          log.warn("No OSGi manifest available to determine version properties");
+      }
+  }
+
+  /** 
+   * Returns whether this is a Brooklyn dev environment,
+   * specifically core/target/classes/ is on the classpath for the org.apache.brooklyn.core project.
+   * <p>
+   * In a packaged or library build of Brooklyn (normal usage) this should return false,
+   * and all OSGi components should be available.
+   */
+  public static boolean isDevelopmentEnvironment() {
+      Boolean isDevEnv = IS_DEV_ENV.get();
+      if (isDevEnv!=null) return isDevEnv;
+      synchronized (IS_DEV_ENV) {
+          isDevEnv = computeIsDevelopmentEnvironment();
+          IS_DEV_ENV.set(isDevEnv);
+          return isDevEnv;
+      }
+  }
+  
+  private static boolean computeIsDevelopmentEnvironment() {
+      Enumeration<URL> paths;
+      try {
+          paths = BrooklynVersion.class.getClassLoader().getResources("brooklyn/BrooklynVersion.class");
+      } catch (IOException e) {
+          // shouldn't happen
+          throw Exceptions.propagate(e);
+      }
+      while (paths.hasMoreElements()) {
+          URL u = paths.nextElement();
+          if (u.getPath().endsWith("core/target/classes/brooklyn/BrooklynVersion.class")) {
+              try {
+                  log.debug("Brooklyn debug environment detected; BrooklynVersion class is at: "+u);
+                  return true;
+              } catch (Exception e) {
+                  Exceptions.propagateIfFatal(e);
+                  log.warn("Error reading manifest to determine whether this is a development environment: "+e, e);
+              }
+          }
+      }
+      return false;
+  }
+
+  public void logSummary() {
+      log.debug("Brooklyn version "+getVersion()+" (git SHA1 "+getSha1FromOsgiManifest()+")");
+  }
+
+  /** @deprecated since 0.7.0, redundant with {@link #get()} */ @Deprecated
+  public static String getVersionFromStatic() {
+      return VERSION_FROM_STATIC;
   }
 
   public static String get() {
-    return INSTANCE.version;
+    return getVersionFromStatic();
```

Comment:
Would prefer this to call `getVersion()`, rather than the deprecated `getVersionFromStatic()`.